kratos/sources/model_part.cpp

```diff
@@ -1594,6 +1594,227 @@ void ModelPart::RemoveConditionsFromAllLevels(Flags IdentifierFlag)
+
+    //add the new geometry
+    this->AddGeometry(p_geometry);
+
+    return p_geometry;
+
+    KRATOS_CATCH("")
+}
+
+ModelPart::GeometryType::Pointer ModelPart::CreateNewGeometry(
+    std::string GeometryTypeName,
+    IndexType GeometryId,
+    std::vector<IndexType> GeometryNodeIds
+    )
+{
+    if (IsSubModelPart()) {
+        GeometryType::Pointer p_new_geometry = mpParentModelPart->CreateNewGeometry(GeometryTypeName, GeometryId, GeometryNodeIds);
+        this->AddGeometry(p_new_geometry);
+        return p_new_geometry;
+    }
+
+    GeometryType::PointsArrayType p_geometry_nodes;
+    for (IndexType i = 0; i < GeometryNodeIds.size(); ++i) {
+        p_geometry_nodes.push_back(pGetNode(GeometryNodeIds[i]));
+    }
+
+    return CreateNewGeometry(GeometryTypeName, GeometryId, p_geometry_nodes);
+}
+
+ModelPart::GeometryType::Pointer ModelPart::CreateNewGeometry(
+    std::string GeometryTypeName,
+    IndexType GeometryId,
+    GeometryType::PointsArrayType pGeometryNodes
+    )
+{
+    KRATOS_TRY
+    
+    if (IsSubModelPart()) {
+        GeometryType::Pointer p_new_geometry = mpParentModelPart->CreateNewGeometry(GeometryTypeName, GeometryId, pGeometryNodes);
+        this->AddGeometry(p_new_geometry);
+        return p_new_geometry;
+    }
+
+    KRATOS_ERROR_IF(this->HasGeometry(GeometryId)) << "Trying to construct an geometry with ID: " << GeometryId << ". A geometry with the same Id exists already." << std::endl;
+
+    // Create the new geometry
+    GeometryType const& r_clone_geometry = KratosComponents<GeometryType>::Get(GeometryTypeName);
+    GeometryType::Pointer p_geometry = r_clone_geometry.Create(GeometryId, pGeometryNodes);
+
+    //add the new geometry
+    this->AddGeometry(p_geometry);
+
+    return p_geometry;
+    
+    KRATOS_CATCH("")
+}
+
+ModelPart::GeometryType::Pointer ModelPart::CreateNewGeometry(
+    std::string GeometryTypeName,
+    IndexType GeometryId,
+    GeometryType::Pointer pGeometry
+    )
+{
+    KRATOS_TRY
+    
+    if (IsSubModelPart()) {
+        GeometryType::Pointer p_new_geometry = mpParentModelPart->CreateNewGeometry(GeometryTypeName, GeometryId, pGeometry);
+        this->AddGeometry(p_new_geometry);
+        return p_new_geometry;
+    }
+
+    KRATOS_ERROR_IF(this->HasGeometry(GeometryId)) << "Trying to construct an geometry with ID: " << GeometryId << ". A geometry with the same Id exists already." << std::endl;
+
+    // Create the new geometry
+    GeometryType const& r_clone_geometry = KratosComponents<GeometryType>::Get(GeometryTypeName);
+    GeometryType::Pointer p_geometry = r_clone_geometry.Create(GeometryId, pGeometry);
+
+    //add the new geometry
+    this->AddGeometry(p_geometry);
+
+    return p_geometry;
+    
+    KRATOS_CATCH("")
+}
+
+ModelPart::GeometryType::Pointer ModelPart::CreateNewGeometry(
+    std::string GeometryTypeName,
+    const std::string& GeometryIdentifierName,
+    std::vector<IndexType> GeometryNodeIds
```

Comment:
```suggestion
    const std::vector<IndexType> GeometryNodeIds
```
Do check other places and arguments where we can add `const`